pennylane/ops/op_math/controlled_class.py

```diff
@@ -0,0 +1,477 @@
+
+        Options are:
+            * `"_prep"`
+            * `"_ops"`
+            * `"_measurements"`
+            * `None`
+        """
+        return self.base._queue_category  # pylint: disable=protected-access
+
+    # Methods ##########################################
+
+    def queue(self, context=QueuingContext):
+        context.safe_update_info(self.base, owner=self)
+        context.append(self, owns=self.base)
+        return self
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        return self.base.label(decimals=decimals, base_label=base_label, cache=cache)
+
+    def matrix(self, wire_order=None):
+
+        base_matrix = self.base.matrix()
+        interface = qmlmath.get_interface(base_matrix)
+
+        num_target_states = 2 ** len(self.target_wires)
+        num_control_states = 2 ** len(self.control_wires)
+        total_matrix_size = num_control_states * num_target_states
+
+        padding_left = self._control_int * num_target_states
+        padding_right = total_matrix_size - padding_left - num_target_states
+
+        left_pad = qmlmath.cast_like(qmlmath.eye(padding_left, like=interface), 1j)
+        right_pad = qmlmath.cast_like(qmlmath.eye(padding_right, like=interface), 1j)
+
+        canonical_matrix = qmlmath.block_diag([left_pad, base_matrix, right_pad])
+
+        if wire_order is None or self.wires == Wires(wire_order):
+            return canonical_matrix
+
+        active_wires = self.control_wires + self.target_wires
+        return operation.expand_matrix(canonical_matrix, wires=active_wires, wire_order=wire_order)
+
+    # pylint: disable=arguments-differ
+    def sparse_matrix(self, wire_order=None, format="csr"):
+        if wire_order is not None:
+            raise NotImplementedError("wire_order argument is not yet implemented.")
+
+        try:
+            target_mat = self.base.sparse_matrix()
+        except operation.SparseMatrixUndefinedError:
+            try:
+                target_mat = sparse.lil_matrix(self.base.matrix())
+            except operation.MatrixUndefinedError as e:
+                raise operation.SparseMatrixUndefinedError from e
+
+        num_target_states = 2 ** len(self.target_wires)
+        num_control_states = 2 ** len(self.control_wires)
+        total_states = num_target_states * num_control_states
+
+        start_ind = self._control_int * num_target_states
+        end_ind = start_ind + num_target_states
+
+        m = sparse.eye(total_states, format="lil", dtype=target_mat.dtype)
+
+        m[start_ind:end_ind, start_ind:end_ind] = target_mat
+
+        return m.asformat(format=format)
+
+    def eigvals(self):
+        base_eigvals = self.base.eigvals()
+        num_target_wires = len(self.target_wires)
+        num_control_wires = len(self.control_wires)
+
+        total = 2 ** (num_target_wires + num_control_wires)
+        ones = np.ones(total - len(base_eigvals))
+
+        return qmlmath.concatenate([ones, base_eigvals])
```

Comment:
I think this will cause issues when computing quantities like expectation values, since the eigenvalues have to ordered in a particular manner which aligns with how the diagonalizing gates map the computational basis states to the Eigen vectors. 

Might be worth trying to compute the expval of `Controlled(qml.PauliZ(0), 1)`, since the control qubit is the 2nd qubit, The Eigen values should have the order: `[1, -1, 1, 1]`, but I think this code will produce `[1, 1, 1, -1]`. Pair this with a circuit prepared in the state `|01>` and you should see two different expectation values.

Reply:
```
>>> qml.CZ(wires=(1,0)).eigvals()
array([ 1,  1,  1, -1])
```
So this is consistent with how it's currently implemented.

Reply:
Wire labels don't have any intrinsic order to them without an additional reference, like device wires.  We tend to use integer wires, but for any wire ordering problem, contemplate completely unordered wires, like `"aux"` and `"wire"`.